src/ensembl/src/shared/hooks/useOutsideClick.tsx

```diff
@@ -17,17 +17,47 @@
 import { useEffect } from 'react';
 
 export default function useOutsideClick<T extends HTMLElement>(
-  ref: React.RefObject<T>,
+  refOrRefs: React.RefObject<T> | React.RefObject<T>[],
   callback: () => void
 ) {
+  const refs = Array.isArray(refOrRefs) ? refOrRefs : [refOrRefs];
+
+  let clickedInside = false;
+
   const handleClickOutside = (event: Event) => {
-    if (ref.current && !ref.current.contains(event.target as HTMLElement)) {
+    let clickedOutside = true;
+
+    if (clickedInside) {
+      // Reset the clickedInside flag to false
+      clickedInside = false;
+      return;
+    }
+
+    refs.forEach((ref) => {
+      if (ref.current && ref.current.contains(event.target as HTMLElement)) {
+        clickedOutside = false;
+      }
+    });
+
+    if (clickedOutside) {
       callback();
     }
   };
 
   useEffect(() => {
+    /*
+      When a child node of the reference node is clicked and is removed from the DOM
+      ref.current.contains(event.target) will return false.
+      To deal with this, we are adding a click event listener to the ref to capture
+      all the clicks to any element within the ref to update the clickedInside flag.
+    */
+    refs.forEach((ref) => {
+      ref?.current?.addEventListener('click', () => {
+        clickedInside = true;
+      });
+    });
+
     document.addEventListener('click', handleClickOutside);
     return () => document.removeEventListener('click', handleClickOutside);
```

Comment:
What about all the event listeners that you added in `refs.forEach`? We should clean them up as well.

Reply:
Would it not get cleaned up automatically?
https://stackoverflow.com/questions/12528049/if-a-dom-element-is-removed-are-its-listeners-also-removed-from-memory/12528067

Reply:
That's assuming the refs passed to this hooks get unmounted when the component that called the hook is unmounted. Which they probably should; but I wonder if there can be any weird edge cases. I we just remove the listeners we won't even have to think about this.

Reply:
That's done